buildSrc/src/main/kotlin/tasks/InstallAndroidSdkTask.kt

```diff
@@ -0,0 +1,94 @@
+package tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.UntrackedTask
+import org.gradle.tooling.GradleConnector
+import java.io.File
+
+/**
+ * Installs Android SDK. This task requires the ANDROID_SDK_ROOT env variable to be set.
+ */
+@UntrackedTask(because = "Output directory can change when running other builds")
+abstract class InstallAndroidSdkTask : DefaultTask() {
+
+    @get:Input
+    abstract val androidSdkVersion: Property<String>
+
+    /**
+     * Optional since we use manual error message
+     */
+    @get:Input
+    @get:Optional
+    abstract val androidSdkRootEnvVariable: Property<String>
+
+    @get:OutputDirectory
+    abstract val androidProjectDir: DirectoryProperty
+
+    @TaskAction
+    fun install() {
+        if (!androidSdkRootEnvVariable.isPresent) {
+            throw GradleException("ANDROID_SDK_ROOT env variable is not set but it should be.")
+        }
+
+        val androidSdkVersion = androidSdkVersion.get()
+        val projectDir = androidProjectDir.get().asFile
+
+        createAndroidProject(projectDir, androidSdkVersion)
+        buildAndroidProject(projectDir)
+    }
+
+    private fun buildAndroidProject(projectDir: File) {
+        val connector = GradleConnector.newConnector()
+            .forProjectDirectory(projectDir)
+        connector.connect().use {
+            it.newBuild().forTasks("build").run()
```

Comment:
I wonder if you'd need to set the environment variable `ANDROID_SDK_ROOT` to `androidSdkRootEnvVariable` for this build, so it actually populates the right directory. I think the other option is to use `System.getenv()` in the task action directly, so you don't depend on anybody configuring the right location.

Reply:
This is not needed, will use System.getenv()